### PR TITLE
Add Test Plugin arista.avd.defined

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Add new custom test plugin to the AVD collection. This will allow refactoring of templates to make them simpler and easier to read.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Component(s) name
ansible-avd collection

## Proposed changes

Adding new test plugin called `arista.avd.defined` which can test a value for `is defined` and `is not None` and optionally also for `== <value>` in a single go.
Example:
```jinja
#A = undefined
B = None
C = "c"
D = "d"

# Jinja test examples:
{% if A is arista.avd.defined %}  =>  false
{% if B is arista.avd.defined %}  =>  false
{% if C is arista.avd.defined %}  =>  true
{% if D is arista.avd.defined %}  =>  true

{% if A is arista.avd.defined("c") %}  =>  false
{% if B is arista.avd.defined("c") %}  =>  false
{% if C is arista.avd.defined("c") %}  =>  true
{% if D is arista.avd.defined("c") %}  =>  false
```

Using this new test we can simplify templates like the following example taken from `ethernet-interfaces.j2`:
### Before:
```jinja
{%             if ethernet_interfaces[ethernet_interface].flowcontrol is defined and ethernet_interfaces[ethernet_interface].flowcontrol is not none %}
{%                 if ethernet_interfaces[ethernet_interface].flowcontrol.received is defined and ethernet_interfaces[ethernet_interface].flowcontrol.received is not none %}
   flowcontrol receive {{ ethernet_interfaces[ethernet_interface].flowcontrol.received }}
{%                 endif %}
{%             endif %}
{%             if ethernet_interfaces[ethernet_interface].mode is defined and ethernet_interfaces[ethernet_interface].mode == "access" %}
   switchport access vlan {{ ethernet_interfaces[ethernet_interface].vlans }}
{%             endif %}
```
### After:
```jinja
{%             if ethernet_interfaces[ethernet_interface].flowcontrol is arista.avd.defined %}
{%                 if ethernet_interfaces[ethernet_interface].flowcontrol.received is arista.avd.defined %}
   flowcontrol receive {{ ethernet_interfaces[ethernet_interface].flowcontrol.received }}
{%                 endif %}
{%             endif %}
{%             if ethernet_interfaces[ethernet_interface].mode is arista.avd.defined("access") %}
   switchport access vlan {{ ethernet_interfaces[ethernet_interface].vlans }}
{%             endif %}
```

## How to test

I have tested with a custom template:
```jinja
{# e1 is undefined #}
{% set e2 = 'e2' %}
!!! Tests
{% if e1 is arista.avd.defined %} e1 has a valid value {% else %} e1 does not have a valid value {% endif %} 
{% if e2 is arista.avd.defined %} e2 has a valid value {% else %} e2 does not have a valid value {% endif %} 
{% if e1 is arista.avd.defined("e1") %} e1 is "e1" {% else %} e1 is not "e1" {% endif %} 
{% if e2 is arista.avd.defined("e2") %} e2 is "e2" {% else %} e2 is not "e2" {% endif %} 
```
Result:
```
!!! Tests
 e1 does not have a valid value  
 e2 has a valid value  
 e1 is not "e1"  
 e2 is "e2"  
``` 
## Checklist:

- [X] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/README.md [hunk 1]
```diff
@@ -70,6 +70,36 @@ text : {{ variable }}
 {% endif %}
 ```
 
+## Plugin Tests
+
+Arista AVD provides built-in filters to help extend jinja2 templates
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/plugins/README.md]: update this to `Arista AVD provides built-in test plugins to help verify data in Jinja2 templates.`

**Review by carlbuchmann — CHANGES_REQUESTED** [review]
- [x] rename test plugin `arista.value` to `arista.avd.defined`
- [x] rename filter plugin `arista.avd.is_valid` to `arista.avd.is_defined` (After validating that is in not in use)
Done

File: ansible_collections/arista/avd/plugins/README.md (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/README.md]: Updated

**Review by titom73 — APPROVED** [review]
Looks good to me and will be more than helpful very shortly!

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Pull request merged**